pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -239,7 +239,7 @@ func convertNetDNSConfigInfo(dnsConfig *vimtypes.NetDnsConfigInfo) *vmopv1.Virtu
 		DHCP:          dnsConfig.Dhcp,
 		DomainName:    dnsConfig.DomainName,
 		HostName:      dnsConfig.HostName,
-		Nameservers:   dnsConfig.IpAddress,
+		Nameservers:   util.Dedupe(dnsConfig.IpAddress),
 		SearchDomains: dnsConfig.SearchDomain,
```

Comment:
Could `SearchDomains` also be duplicated by any chance?

Reply:
Not observing this during tests but it's a valid concern.